File: medis-service/src/main/java/io/mersys/medis/documents/patient/Diseases.java

package io.mersys.medis.documents.patient;

import java.time.LocalDate;

import javax.validation.constraints.NotBlank;

import org.springframework.data.annotation.TypeAlias;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
@TypeAlias("Diseases")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Diseases {

	@NotBlank
	private String diseasesName;

	private LocalDate startDate;

	private LocalDate endDate;
	
	private String description;

}

File: medis-ui/src/app/main/patients/classes/diseases.ts
export class Diseases{
  constructor(public diseaseName?: string, public startDate?: Date, public endDate?: Date, public description?: string){
    this.diseaseName = diseaseName;
    this.startDate = startDate;
    this.endDate = endDate;
    this.description = description;
  }

}
